rfcs/2021-11-whcm-support.md

```diff
@@ -0,0 +1,108 @@
+    - It's not entirely obvious an error has happened in a form field without reading the text
+  - The solution is to avoid relying on color alone when conveying different messages/states (_needs a11y/design direction_)
+    - Different border styles/widths?
+    - Use of italic or bold text?
+    - Label error text with "Error:" language?
+  - Seen in all form fields, but mostly inherited from the `FormLabel` component
+- Icons vary in visibility
+  - Some icons don't render
+  - Icons that do render don't reflect WHCM color palette and are difficult to see
+  - Reliance on icons only to convey meaning is lost, especially for Alerts, Buttons with icons, and displaying active states in navigation
+  - Icons should inherit the appropriate text color as mapped to the WHCM theme
+- Border issues
+  - Thin/no borders make it hard to see Badge, FilterChip, HelpDrawerToggle, Tooltip, and the Modal close button
+  - Content in Accordion and Tabs aren't visually grouped together as they are in non-WHCM
+- Buttons (_needs a11y/design direction_)
+  - Use of "ghost style" buttons results in weird padding/margins and/or broken borders
+  - Should buttons and links styled as buttons look the same?
+
+## Proposal
+
+CSS media queries exist to detect if WHCM is enabled and can help developers modify how its styles are applied to specific elements.
+
+`[-ms-high-contrast](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/-ms-high-contrast)` is a Microsoft extension that targets WHCM for IE and older versions of Edge. It supports `none`, `active`, `black-on-white`, and `white-on-black` as values. But it's recommended at this point to only use the `active` value.
+
+`[forced-colors](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/forced-colors)` is a new media query supported by Chrome, Edge, and Firefox. It supports `none` and `active` as values. It should also be noted that this media feature spec is actively being worked on.
+
+[Windows recommends using both media queries](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/) (aka ["The Frankenquery"](https://adrianroselli.com/2021/02/whcm-and-system-colors.html#Frankenquery)) to guarantee adequate support for those using WHCM. This would look something like:
+
+```css
+@media (-ms-high-contrast: active), (forced-colors: active) {
+  /* CSS overrides here */
+}
+```
+
+While our design system no longer technically supports IE (as [it falls below our 2% support threshold](https://analytics.usa.gov/health-human-services/)), a good percentage of those who rely on WHCM continue to use it. And because the effort to support IE is low, I agree with Microsoft's recommendation - just be certain your system color keywords work across both queries and the cascade doesn't impact how those styles render (meaning, test the changes). If more targeted styles are needed between IE and Edge, this query may need to be split.
```

Comment:
Another thing I might mention is forced colors is a standard, meaning that Chrome, Safari, and Firefox should hypothetically also all roll out support. It'd be good to have our content in good shape for if and when that happens.

I think Chrome in particular will do it sooner than later, as they tend to hoover up all of Edge's Webkit-level accessibility work.